cloudify_agent/api/plugins/installer.py

```diff
@@ -350,81 +351,83 @@ def _rmtree(path):
 
 def extract_package_to_dir(package_url):
     """
-    Extracts a pip package to a temporary directory.
-
-    :param package_url: the URL to the package source.
-
+    Using a subprocess to extracts a pip package to a temporary directory.
+    :param: package_url: the URL to the package source.
     :return: the directory the package was extracted to.
-    """
-
-    # 1) Plugin installation during deployment creation occurs not in the main
-    # thread, but rather in the local task thread pool.
-    # When installing source based plugins, pip will install an
-    # interrupt handler using signal.signal, this will fail saying something
-    # like "signals are only allowed in the main thread"
-    # from examining the code, I found patching signal in pip.util.ui
-    # is the cleanest form. No "official" way of disabling this was found.
-    # 2) pip.utils logger may be used internally by pip during some
-    # ImportError. This interferes with our ZMQLoggingHandler which during
-    # the first time it is invoked tries importing some stuff. This causes
-    # a deadlock between the handler lock and the global import lock. One side
-    # holds the import lock and tried writing to the logger and is blocked on
-    # the handler lock, while the other side holds the handler lock and is
-    # blocked on the import lock. This is why we patch the logging level
-    # of this logger - by name, before importing pip. (see CFY-4866)
-    _previous_signal = []
-    _previous_level = []
-
-    def _patch_pip_download():
-        pip_utils_logger = logging.getLogger('pip.utils')
-        _previous_level.append(pip_utils_logger.level)
-        pip_utils_logger.setLevel(logging.CRITICAL)
-
-        try:
-            import pip.utils.ui
-
-            def _stub_signal(sig, action):
-                return None
-            if hasattr(pip.utils.ui, 'signal'):
-                _previous_signal.append(pip.utils.ui.signal)
-                pip.utils.ui.signal = _stub_signal
-        except ImportError:
-            pass
-
-    def _restore_pip_download():
-        try:
-            import pip.utils.ui
-            if hasattr(pip.utils.ui, 'signal') and _previous_signal:
-                pip.utils.ui.signal = _previous_signal[0]
-        except ImportError:
-            pass
-        pip_utils_logger = logging.getLogger('pip.utils')
-        pip_utils_logger.setLevel(_previous_level[0])
 
+    """
     plugin_dir = None
+    archive_dir = tempfile.mkdtemp()
+    runner = LocalCommandRunner()
+
     try:
-        plugin_dir = tempfile.mkdtemp()
-        _patch_pip_download()
-        # Import here, after patch
-        import pip
-        pip.download.unpack_url(link=pip.index.Link(package_url),
-                                location=plugin_dir,
-                                download_dir=None,
-                                only_download=False)
-    except Exception as e:
+        command = [get_pip_path(), 'download', '-d',
```

Comment:
Let's add some more context here - why are we doing it this way, why we can't just use `curl`, etc.